PROVESFlightControllerReference/test/int/bridge_uplink_test.py

```
@@ -0,0 +1,471 @@
+    print(f"[await_dl] TIMEOUT: {dest} not complete in {secs}s", flush=True)
+
+
+@pytest.mark.parametrize(
+    "src", [os.environ.get("UPLINK_SRC", "/tmp/grc_scratch_2k.bin")]
+)
+def test_uplink_file(fprime_test_api: IntegrationTestAPI, src):
+    """Uplink a file over the LoRa bridge and await FileUplink completion."""
+    dest = os.environ.get("UPLINK_DEST", "//bridge_test.bin")
+    timeout = int(os.environ.get("UPLINK_TIMEOUT", "120"))
+    with open(src, "rb") as f:
+        digest = hashlib.md5(f.read()).hexdigest()
```

Ruff (0.15.20)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)